apps/web/src/components/Feed/Posts/PostCreatorAndCollectionSection.tsx

```diff
@@ -45,20 +40,44 @@ export function PostCreatorAndCollectionSection({ tokenRef }: Props) {
   );
 
   const { contractName } = extractRelevantMetadataFromToken(token);
+  const contractNameCharCount = contractName.length;
+  const creatorUsernameCharCount = useMemo(() => {
+    if (token.community?.creator) {
+      if (token.community.creator.__typename === 'GalleryUser') {
+        return token.community.creator.username?.length ?? 0;
+      }
+    }
+    return 0;
+  }, [token.community?.creator]);
 
-  const CreatorComponent = useMemo(() => {
-    if (token.owner && token.ownerIsCreator) {
-      return (
-        <UserHoverCard userRef={token.owner}>
-          <OwnerProfilePictureAndUsername
-            userRef={token.owner}
-            eventContext={contexts.Posts}
-            pfpDisabled
-          />
-        </UserHoverCard>
-      );
+  const containerStyles = useMemo(() => {
+    let collectionWidth = 66;
+    let creatorWidth = 33;
+    if (
+      contractNameCharCount > LONG_NAME_CHAR_BREAKPOINT &&
+      creatorUsernameCharCount > LONG_NAME_CHAR_BREAKPOINT
+    ) {
+      collectionWidth = 50;
+      creatorWidth = 50;
+    } else if (creatorUsernameCharCount === 0) {
+      return null;
+    } else if (
+      contractNameCharCount < LONG_NAME_CHAR_BREAKPOINT &&
+      creatorUsernameCharCount < LONG_NAME_CHAR_BREAKPOINT
+    ) {
+      // space-between styling applied separately on parent container
+      return null;
+    } else if (contractNameCharCount > LONG_NAME_CHAR_BREAKPOINT) {
+      collectionWidth = 33;
+      creatorWidth = 66;
     }
+    return {
+      collectionWidth: collectionWidth,
+      creatorWidth: creatorWidth,
+    };
+  }, [contractNameCharCount, creatorUsernameCharCount]);
 
+  const CreatorComponent = useMemo(() => {
     if (token.community?.creator) {
       if (token.community.creator.__typename === 'GalleryUser') {
```

Comment:
same comment on consolidation.

in fact i vote we deprecate defining `CreatorComponent` in `useMemo` and just add a ternary in the render logic directly:

```
{token.community?.creator?.__typename === 'GalleryUser' ? <UserHoverCard ... /> : null}
```

feel free to apply in above files where it makes sense